tsconfig.base.json

```diff
@@ -15,7 +15,7 @@
     "noPropertyAccessFromIndexSignature": true,
     "noUncheckedIndexedAccess": true,
     "noUnusedLocals": true,
-    "noUnusedParameters": true,
+    "noUnusedParameters": false,
```

Comment:
I've figured out a way to get this working. For the virtual classes, just overload the function. For example, instead of
```ts
class {
  fakeMethod(value: string) { throw new Error('unimplemented'); }
}
```
use

```ts
class {
  fakeMethod(value: string): string;
  fakeMethod() { throw new Error('unimplemented'); }
}
```
In TS, if there is one overload, that overload becomes the signature, not the one below it.

Reply:
@jrandolf does the doc generator properly generated docs for this?

Reply:
I mean the following works too:

```
class {
  fakeMethod(value: string) { value; throw new Error('unimplemented'); }
}
```

But I think disabling the rule is the simplest since we still have checking in the IDE.

Reply:
I'm not sure, but it should since TS doesn't even consider the method below the first overload as the public interface.